app/controllers/demands_controller.rb

```diff
@@ -24,9 +26,34 @@ def show
     @demand = Demand.find(params[:id])
   end
 
+  def edit
+    @demand = Demand.find(params[:id])
+  end
+
+  def update
+    form = DemandForm.new(current_user, update_params.merge({id: params[:id]}))
+
+    if form.save
+      redirect_to(Demand.find(params[:id]))
+    else
+      flash[:alert] = "Oops — we couldn't save those changes..."
+      render :edit
+    end
+  end
+
   private
 
+  def require_permission
+    if !current_user_owns_demand?(Demand.find(params[:id]))
```

Comment:
`Demand.find` will raise a 404 if it can't be found. Not necessarily something to change but just wanted to mention.